test/e2e/webdriver/driver.js

```diff
@@ -1000,6 +1004,46 @@ class Driver {
     throw new Error(`No window with url: ${url}`);
   }
 
+  /**
+   * Waits until the current URL matches the specified URL.
+   *
+   * @param {string} url - URL to wait for.
+   * @param {int} delayStep - optional delay between retries, defaults to 1000 milliseconds.
+   * @param {int} timeout - optional timeout period, defaults to this.timeout.
+   * @param {int} retries,retryDelay - optional number of retries for the URL fetch operation, defaults to 8.
+   * @returns {Promise<void>} Promise that resolves once the URL matches.
+   * @throws {Error} Throws an error if the URL does not match within the timeout period.
+   */
+  async waitForUrl(
+    url,
```

Comment:
Using an object for these parameters would be advisable as well. Especially in this case where so many are optional, so we don't have to call it with `undefined` in various places (we actually have a lint rule against this, though it doesn't appear to be enabled in this repo).;

Reply:
thank you, that's again a great point. I've updated the function :pray: